File: FE/NEAprojectFE/src/schemas/receiver.ts
import * as z from 'zod'

export const createReceiverSchema = (t: (key: string) => string) => {
    return z.object({
        name: z.string().min(1, t("createReceiver.errors.nameRequired")),
        office_name: z.string().min(1, t("createReceiver.errors.deptRequired")),
        office_address: z.string().min(1, t("createReceiver.errors.deptAddressRequired")),
        id_card_number: z.string().min(1, t("createReceiver.errors.idNo")),
        id_card_type: z.enum(["national_id", "citizenship", "voter_id", "passport", "drivers_license", "pan_card", "unknown"], t("createReceiver.errors.idType")),
        phone_number: z.string()
            .min(1, t("createReceiver.errors.phone"))
            .regex(/^[\d\u0966-\u096F]+$/, t("createReceiver.errors.phoneNum"))
            .max(10, t("createReceiver.errors.phoneMax")),
        post: z.string().min(1, t("createReceiver.errors.post")),
        vehicle_number: z.string().min(1, t("createReceiver.errors.vehicleNo")),
    })
}

export const updateReceiverSchema = (t: (key: string) => string) => {
    return z.object({
        name: z.string().min(1, t("editReceiver.errors.nameRequired")),
        office_name: z.string().min(1, t("editReceiver.errors.deptRequired")),
        office_address: z.string().min(1, t("editReceiver.errors.deptAddressRequired")),
        id_card_number: z.string().min(1, t("editReceiver.errors.idNo")),
        id_card_type: z.enum(["national_id", "citizenship", "voter_id", "passport", "drivers_license", "pan_card", "unknown"], t("editReceiver.errors.idType")),
        phone_number: z.string()
            .min(1, t("editReceiver.errors.phone"))
            .regex(/^[\d\u0966-\u096F]+$/, t("editReceiver.errors.phoneNum"))
            .max(10, t("editReceiver.errors.phoneMax")),
        post: z.string().min(1, t("editReceiver.errors.post")),
        vehicle_number: z.string().min(1, t("editReceiver.errors.vehicleNo")),
    })
}

File: BE/NEAProjectBE/myapp/models/receiver.py
from django.db import models
from .base import TimeStampedModel


class Receiver(TimeStampedModel):
    class IDCardType(models.TextChoices):
        NATIONAL_ID = "national_id", "राष्ट्रिय परिचयपत्र"
        CITIZENSHIP = "citizenship", "नागरिकता प्रमाणपत्र"
        VOTER_ID = "voter_id", "मतदाता परिचयपत्र"
        PASSPORT = "passport", "राहदानी / ई–राहदानी"
        DRIVERS_LICENSE = "drivers_license", "सवारी चालक अनुमति पत्र"
        PAN_CARD = "pan_card", "स्थायी लेखा नम्बर (प्यान)"
        UNKNOWN = "unknown", "अज्ञात"
        EMPLOYEE_ID = "employee_id", "कर्मचारी परिचयपत्र"

    name = models.CharField(max_length=255)
    post = models.CharField(max_length=255, default="UNKNOWN")
    id_card_number = models.CharField(max_length=50, default="UNKNOWN")
    id_card_type = models.CharField(
        max_length=20, choices=IDCardType.choices, default=IDCardType.UNKNOWN
    )
    office_name = models.CharField(max_length=255, default="UNKNOWN")
    office_address = models.TextField(default="UNKNOWN")
    phone_number = models.CharField(max_length=20, default="UNKNOWN")
    vehicle_number = models.CharField(max_length=50, default="UNKNOWN")

    def __str__(self):
        return self.name

    class Meta:
        app_label = "myapp"

